src/main/java/com/sap/oss/phosphor/fosstars/model/value/ArtifactVersion.java

```diff
@@ -0,0 +1,132 @@
+package com.sap.oss.phosphor.fosstars.model.value;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.core.JsonGenerator;
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.SerializerProvider;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+import com.fasterxml.jackson.databind.deser.std.StdDeserializer;
+import com.fasterxml.jackson.databind.ser.std.StdSerializer;
+import java.io.IOException;
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Contains version tag and release date for an artifact version.
+ */
+public class ArtifactVersion {
+
+  public static final ArtifactVersion EMPTY = new ArtifactVersion("", LocalDate.now());
+
+  private final String version;
+  @JsonDeserialize(using = LocalDateDeserializer.class)
+  @JsonSerialize(using = LocalDateSerializer.class)
+  private final LocalDate releaseDate;
+
+  @JsonIgnore
+  private Optional<SemanticVersion> semanticVersion;
+
+  /**
+   * Initialize the ArtifactVersion based on version tag and release date.
+   *
+   * @param version version tag
+   * @param releaseDate release date
+   */
+  public ArtifactVersion(@JsonProperty("version") String version,
+      @JsonProperty("releaseDate") LocalDate releaseDate) {
+
+    Objects.requireNonNull(version, "Version must be set");
+    Objects.requireNonNull(releaseDate, "Release date must be set");
+
+    this.version = version;
+    this.releaseDate = releaseDate;
+    initSemVer(version);
+  }
+  
+  private void initSemVer(String version) {
+    semanticVersion = SemanticVersion.parse(version);
+  }
+
+  public SemanticVersion getSemanticVersion() {
+    return semanticVersion.orElse(null);
+  }
```

Comment:
`Optional` is a good thing to avoid NPEs. Let's use it.

```suggestion
  public Optional<SemanticVersion> getSemanticVersion() {
    return semanticVersion;
  }
```